frontend/src/utilities/useWatchNotebookEvents.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { getNotebookEvents } from '../services/notebookEventsService';
+import { useDeepCompareMemoize } from './useDeepCompareMemoize';
+import useNotification from './useNotification';
+import { K8sEvent } from '../types';
+
+export const useWatchNotebookEvents = (
+  projectName: string,
+  notebookName: string,
+  watch: boolean,
+): K8sEvent[] => {
+  const [notebookEvents, setNoteBookEvents] = React.useState<K8sEvent[]>([]);
+  const notification = useNotification();
+
+  React.useEffect(() => {
+    let watchHandle;
+    let cancelled = false;
+
+    const clear = () => {
+      cancelled = true;
+      if (watchHandle) {
+        clearTimeout(watchHandle);
+      }
+    };
+
+    const watchNotebookEvents = () => {
+      if (projectName && notebookName && watch) {
+        getNotebookEvents(projectName, notebookName)
+          .then((data: K8sEvent[]) => {
+            setNoteBookEvents(data);
+          })
+          .catch((e) => {
+            notification.error('Error fetching notebook events', e.response.data.message);
+            clear();
+          });
+        watchHandle = setTimeout(watchNotebookEvents, 1000);
+      }
+    };
+
+    if (!cancelled) {
+      watchNotebookEvents();
+    }
+    return clear;
+  }, [projectName, notebookName, notification, watch]);
+
+  const retNotebookEvents = useDeepCompareMemoize<K8sEvent[]>(notebookEvents);
+
+  return retNotebookEvents || [];
```

Comment:
With my change in #333 you don't need to fallback anymore. Stick `value` into the `useRef` default value.

```
const ref = React.useRef<T>(value);
```
eg: https://github.com/opendatahub-io/odh-dashboard/pull/333/files#diff-8fd4c90590d507eb02de02c93e751baea9aa65b6b1bc7402ff8633f289a3db12R5

Reply:
Will updated after rebase.

Reply:
The code will not change, you're welcome to apply the same change I did (see the link above) and git will ignore the "conflict" as it's a 1:1 match. The file is small and unlikely to change after this (likely ever) so it's safe to match.